### PR TITLE
[business-logic] Add first section of business logic

### PR DESCRIPTION
**Pull request by jaylees14**
Add the first part of the "business logic" section, explains where to write code and how to define an empty hook.

### CODE REVIEW

File: TempleEight/docs/guide/business-logic.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+config.json
+dao
+example-service.go
+go.mod
+hook.go
+setup.go
+util
+```
+
+We're most interested in `setup.go` for this guide, which is where you can add additional logic without losing them if you need to regenerate your Templefile.
```
Comment (Lewiky) [inline TempleEight/docs/guide/business-logic.md:26]:
```suggestion
We're most interested in `setup.go` for this guide, which is where you can add additional logic without losing it if you need to regenerate your Templefile.
```

Comment (Lewiky) [reply TempleEight/docs/guide/business-logic.md:26]: ... logic is singular right? I'm suddenly unsure

File: TempleEight/docs/guide/business-logic.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+func (env *env) setup(router *mux.Router) {
+	// Add user defined code here
+}
+```
+
+The `setup` method defined on `env` here is invoked before the HTTP server is started.
+It gives you, the developer, the opportunity to:
+
+- register hooks, to be executed before or after any database calls
```
Comment (Lewiky) [inline TempleEight/docs/guide/business-logic.md:43]: Might be worth a sentence or so just explaining what is meant by a hook

File: TempleEight/docs/guide/business-logic.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
+---
+id: business-logic
+title: Adding business logic to your service 
+sidebar_label: Business Logic 
+---
+
+We know that the code generated by Temple won't necessarily satisfy every use case.
+That's why we made it easy to modify and extend the application logic, without sacrificing the ability to regenerate code after.
+This short guide will walk you through how to add custom business logic to your application, as well as additional endpoints.
+We'll be using the `ExampleProject` from the [getting started guide](getting-started.md).
```
Comment (Lewiky) [inline TempleEight/docs/guide/business-logic.md:10]: Might be worth saying that this tutorial assumes some level of familiarity with writing Go

**Comment by jaylees14** [comment]
Nice, I'll add a diagram in my next PR 😄

**Pull request merged**